stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/BaseChannelListItemAdapter.kt

```diff
@@ -0,0 +1,37 @@
+package io.getstream.chat.android.ui.channel.list.adapter
+
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.ui.channel.list.ChannelListView
+import io.getstream.chat.android.ui.channel.list.ChannelListViewStyle
+import io.getstream.chat.android.ui.channel.list.adapter.diff.ChannelListDiffCallback
+import io.getstream.chat.android.ui.channel.list.adapter.viewholder.BaseChannelListItemViewHolder
+
+public abstract class BaseChannelListItemAdapter :
+    RecyclerView.Adapter<BaseChannelListItemViewHolder>(),
```

Comment:
Why don't we use a [`RecyclerView.ListAdapter`](https://developer.android.com/reference/androidx/recyclerview/widget/ListAdapter) that already implement all logic about replace items and use the diffUtil Callback?